bool req
---
string image_path